indra/newview/gltf/asset.cpp

```diff
@@ -1083,37 +889,76 @@ void Image::decompose(Asset& asset, const std::string& folder)
         }
         else
         {
+            error = "Unknown mime type, saved as .bin";
             extension = ".bin";
         }
 
-        std::string filename = folder + "/" + name + "." + extension;
+        std::string filename = folder + delim + name + extension;
 
         // set URI to non-j2c file for now, but later we'll want to reference the j2c hash
-        mUri = name + "." + extension;
+        mUri = name + extension;
```

Comment:
`mUri = name + extension;`

So if you move that code above line 896 you could:

`std::string filename = folder + delim + mUri;`